### ISSUE TITLE
This should be core

### ISSUE THREAD

**Issue by tractorcow** [lozcalver/silverstripe-stricttemplates#1]
I would be happy to require that templates in 4.0.0 followed strict rules. The lax rules have been deprecated in 3.x for a while, so I see no reason why we should maintain them.

Do you feel like turning this into a pull request?

**Comment by jonom** [comment]
:+1: from me!

The one concern I have from looking at the ReadMe is `strict_braces_mode`. If this preference affects template parsing in all modules then all templates used in core and public modules would be forced to use the `{$var}`syntax to ensure compatibility. Not a deal breaker but just a consideration.

**Comment by lozcalver** [comment]
Yeah that `strict_braces_mode` thing is just personal preference that I put in there because I could :D.

I think something like this should probably be posted to the dev list for discussion. This module in its current form pretty-much opens up the ability to write raw PHP in templates in certain places ([example 1](https://github.com/kinglozzer/silverstripe-stricttemplates/blob/master/tests/StrictTemplateTest.php#L186-L193), [example 2](https://github.com/kinglozzer/silverstripe-stricttemplates/blob/master/tests/StrictTemplateTest.php#L186-L193)) and I suppose there’s a point at which you may as well just be writing the templates in PHP. My other concern would be confusing newbies or people not familiar with PHP by forcing them to quote things in templates. All that said, the current setup is holding us back in places, and things like unquoted strings/non-dollar-marked lookups have been effectively deprecated for a while...

If we’re doing any significant work on the template parser, I’d like to see tests on the actual _parser_ and syntax (our current tests are patchy and all seem to be based on `SSViewer` rather than the parser itself).

**Comment by jonom** [comment]
While we don't want to promote it as a pattern I don't think the ability to write raw PHP in templates is necessarily a bad thing. Would come down to developer discipline I guess. Personally I would only put raw PHP in templates a a last resort but it could be useful to have the option.

I don't think there's a problem forcing newbies to quote things - if anything I think it will help them learn good coding practices. If you're working on a SilverStripe template file you're comfortable with html and probably already quoting your HTML attributes. I imagine most front-end devs today have some experience with javascript too. _Not_ having to quote your function arguments is confusing :)

But yes, should probably have some discussion on the dev list about it. I'll post something in a minute.

**Comment by silbinarywolf** [comment]
@tractorcow @kinglozzer Bump!

Definitely look into adding this into core!

**Comment by michalkleiner** [comment]
@tractorcow @kinglozzer 2018 bump again :-D

**Comment by tractorcow** [comment]
Oh hi 2018. Maybe this is a good hackday project?

**Comment by Taitava** [comment]
Need a volunteer perhaps? I have no experience about the internal working of SilverStripe's template engine so far, but perhaps I could try to do something :). Just let me know what exactly needs to be done :).
